README.md

```diff
@@ -1 +1,116 @@
+
+About
+-----
+This contains a series of [Jasmine]1.3 helpers when using [Jasmine-Node] to facilitate testing certain functionality more easily. These help to detect when a test fails but should have passed, turn middleware into a promise and makes it easier to use promises in tests. More explanation is below for each helper.
+
+*Note: If you are not using [Jasmine-Node] these helpers aren't going to help your testing suite very much.*
+
+How to Use
+----------
+Include the package in your `package.json` file.
+
+```
+npm install --save-dev node-test-helpers
+```
+
+Then you need to be able to include the helper in your testing directory or where you'd like to run tests.
+
+You can do this by creating a symbolic link to files contained. You'll want to link to the files and not the folder. [Jasmine] doesn't seem to pick up on the files if only the folder is linked. Below is an example of setting up a symbolic link. Make sure to include the folder in your `.gitignore` file so you don't commit the files as this folder will show up in your status.
+
+```
+// Linux/MacOS
+mkdir ./path/to/project/test-folder/jasmine-node-helpers
+
+ln -s /path/to/project/from/root/node_modules/jasmine-node-helpers/lib/* ./path/to/project/test-folder/jasmine-node-helpers/
+
+// Windows
+// If you can figure out how to make symlink work, good on you, make a [fork](CONTRIBUTING.md) and update those instructions please.
+```
+
+jasmine.fail
+------------
+This is most useful when testing a promise which you are wanting to check the test is getting to the reject successfully, but want to fail in case code changes which makes the promise resolve instead of reject. This will make the test fail so it can be caught and code can be diagnosed.
+
+```js
+describe("a suite", () => {
+    it("a test", () => {
+        return someClass.someMethodAsync().then(jasmine.fail, () => {
+            expect(someVar).toBe("this");
+        });
+    });
+});
+```
+
+jasmine.middlewareToPromise
+---------------------------
+Used when wanting to make a piece of middleware used in setting up a node server and you want to make it to be run synchronously with your tests to get a response back straight aways without waiting for a real promise to resolve.
+
+This a bit of a complicated process and your middleware will need to be written in the style as the example and test is in, but should make it easy to test your middleware.
+
+```js
+describe("a file", () => {
+    var mock, mockDependency;
+
+    beforeEach(() => {
+        mockDependency = require("./mock/mockDependency");
+        mock = require("mock")(mockDependency);
+    });
+    describe("a suite", () => {
+        it("a test", () => {
+            var middlewareAsync;
+
+            middlewareAsync = jasmine.middlewareToPromise(mock(passedInArgument));
+            middlewareAsync().then(() => {
+                expect(mockDependency.call).toHaveBeenCalled();
+            });
+        });
+    });
```

Comment:
Example seems a bit weird.  Not everyone uses factories, especially a middleware factory factory.

    var middleware, middlewareAsync, req, res;

    middleware = require("../lib/your-favorite-middleware");
    middlewareAsync = jasmine.middlewareToPromise(middleware);

    beforeEach(() => {
        req = makeFakeRequestObject();  // You write this
        res = makeFakeResponseObject();  // You write this
    });
    describe("a suite", () => {
        it("is easier with promises", () => {
            // If an argument is passed, the promise is rejected.
            // If the middleware throws, the promise is rejected.
            // The promise helper automatically handles both.
            // Just remember to `return` the promise.
            return middlewareAsync(req, res).then(() => {
                expect(res.send).toHaveBeenCalled();
            });
        });
        it("works in a traditional way", (done) => {
            // You must manually ensure the middleware does not throw.
            // Also you must manually ensure no argument was passed to
            // the `done` / `next` callback.
            // Don't forget to call `done()` when everything's complete.
            function doneWithMiddleware(arg) {
                expect(arg).not.toBeDefined();
                expect(res.send).toHaveBeenCalled();
                done();
            }
            jasmine.expect(() => {
                middleware(req, res, doneWithMiddleware);
            }).not.toThrow();
        });
    });